crates/ruff_linter/src/rules/pylint/rules/unnecessary_dunder_call.rs

```diff
@@ -124,21 +124,30 @@ pub(crate) fn unnecessary_dunder_call(checker: &mut Checker, call: &ast::ExprCal
                 title = Some(message.to_string());
             }
             ([arg], DunderReplacement::Operator(replacement, message)) => {
-                fixed = Some(format!(
-                    "{} {} {}",
-                    checker.locator().slice(value.as_ref()),
-                    replacement,
-                    checker.locator().slice(arg),
-                ));
+                let value_slice = checker.locator().slice(value.as_ref());
+                let arg_slice = checker.locator().slice(arg);
+
+                if arg.is_attribute_expr() || arg.is_name_expr() || arg.is_literal_expr() {
```

Comment:
I think we can exclude some more expressions here

* Calls
* Unary expressions 
* Lambda
* If expressions (I think)? 
* Dict / Set / List / Tuple and all comprehension variants (everything with parentheses)
* Generators
* Subscripts
* Starred (I think)
* Slices?

Reply:
Reworked it, added all of the above except Unary (because that's what got us this PR in the first place hah)